lndsigner.go

```diff
@@ -223,3 +227,63 @@ func parseNetwork(addr net.Addr) string {
 func ListenOnAddress(addr net.Addr) (net.Listener, error) {
 	return net.Listen(parseNetwork(addr), addr.String())
 }
+
+// GetAccounts is currently used in integration testing, but will soon also
+// be used in policy enforcement. For current status, see the branch at
+// https://github.com/bottlepay/lndsigner/tree/offchain-ratelimiting
+func GetAccounts(acctList string) (map[[3]uint32]string, error) {
+	accounts := make(map[[3]uint32]string)
+
+	elements := make(map[string]interface{})
+
+	err := json.Unmarshal([]byte(acctList), &elements)
+	if err != nil {
+		return nil, err
+	}
+
+	acctElements, ok := elements["accounts"].([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("no accounts returned in JSON")
+	}
+
+	for _, interEl := range acctElements {
```

Comment:
I think I would feel slightly more comfortable if we unmarshalled into a typed struct rather than typecasting on map elements.

Reply:
Fixed in https://github.com/bottlepay/lndsigner/compare/c8d4e2145dcf4ce6412f5b53ebd0f5b7422f79f4..2471acc1aef216941024a860638e0b80789ab598 - that really simplified things